brewcop.py

```diff
@@ -217,64 +205,238 @@ class Brewcop:
+        """
+        Force screen redraw.
+        It normally redraws when control returns to event loop.
+        """
+        self.main_loop.draw_screen()
+
+    @property
+    def headC(self):
+        """Get text from header, center region"""
+        return self._headC.get_text()
+
+    @headC.setter
+    def headC(self, value):
+        """Set text in header, center region"""
+        self._headC.set_text(value)
 
     @property
-    def indicator(self):
-        """Get the indicator text (upper right in header)"""
-        self._indicator.get_text()
+    def headR(self):
+        """Get text from header, right region"""
+        return self._headR.get_text()
 
-    @indicator.setter
-    def indicator(self, value):
-        """Set the indicator text (upper right in header)"""
-        self._indicator.set_text(value)
+    @headR.setter
+    def headR(self, value):
+        """Set text in header, right region"""
+        self._headR.set_text(value)
 
     @property
     def meter(self):
         """Get the meter text (scale reading)"""
-        self._meter.get_text()
+        return self._meter.get_text()
 
     @meter.setter
     def meter(self, value):
         """Set the meter text (scale reading)"""
         self._meter.set_text(value)
 
+    def online(self):
+        """Set online display mode (show background + footer progress bar)"""
+        self.layout.body = self.background
+        self.layout.footer = self.pbar
+
+    def offline(self):
+        """Set offline display mode (show meter + footer message)"""
+        self.layout.body = self.meterbody
+        self.layout.footer = self.footmsg
+
+    def progress(self, value):
+        """Update progress bar value (pot contents in mL)"""
+        self.pbar.set_completion(value)
+
+
+class Brains:
+    """
+    Add some scale memory and semantics for interpreting a series
+    of weights as human activity.
+
+    Implement a state machine consisting of the following states:
+    unknown - no scale readings stored yet
+    brewing - scale readings have shown some increase over last 30s
+    ready - scale readings are stable/decreasing and pot still has content
+    empty - scale readings are stable/decreasing and pot content is low
+    """
+
+    """Retain scale samples for history_length seconds"""
+    history_length = 30
+
+    def __init__(self, tick_period=1, empty_thresh=0, stale_thresh=60*60*8):
+        self.history = deque(maxlen=int(self.history_length / tick_period))
+        self.pot_empty_thresh_g = empty_thresh
+        self.stale_thresh = stale_thresh
+        self.state = "unknown"
+        self.timestamp = 0
+
+    def notify(self):
+        """Stub for slack notification"""
+        return
+
+    def increasing(self, l):
+        """
+        Return true if any of l[i] > l[i + 1].
+        N.B. 'l' must be a list as deques don't allow slicing
+        N.B. Ignores l[i] < l[i + 1].
+        """
+        return any(x > y for x, y in zip(l, l[1:]))
```

Comment:
Since this is only ever called on `self.history`, maybe remove `l` as an argument and add `l = list(self.history)`?
Or maybe remove it as a method and instead make it a standalone function?

There is also [`itertools.islice`](https://docs.python.org/2/library/itertools.html#itertools.islice)

Reply:
I'll just drop the argument for now, if that's OK.